### PR TITLE
fix: offcanvas touch scrolling

### PR DESCRIPTION
**Pull request by rastislav-chynoransky**
## Description

In touch events `pageY` is not direct property of an event (the fixed methods are used only in touch-enabled context). Also the detection of scrolling direction was reversed.

This causes stucked offcanvas menu when scrolled to the bottom.
Seen also on https://get.foundation/sites/docs/

Tested with Chrome for Android

- Closes (no issue)

## Types of changes

- [ ] Documentation
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (anything that would change an existing functionality)
- [ ] Maintenance (refactor, code cleaning, development tools...)

## Checklist

- [x] I have read and follow the CONTRIBUTING.md document.
- [x] The pull request title and template are correctly filled.
- [x] The pull request targets the right branch (`develop` or `develop-v...`).
- [x] My commits are correctly titled and contain all relevant information.
- [ ] I have updated the documentation accordingly to my changes (if relevant).
- [ ] I have added tests to cover my changes (if relevant).

### CODE REVIEW

**Comment by DanielRuf** [comment]
> This causes stucked offcanvas menu when scrolled to the bottom.
> Seen also on [get.foundation/sites/docs](https://get.foundation/sites/docs/)

Hi, we use an older Foundation Sites version on the docs, so only some codepen makes sense here ;-)

File: js/foundation.offcanvas.js [hunk 1]
```diff
@@ -364,9 +364,9 @@ class OffCanvas extends Plugin {
   _stopScrollPropagation(event) {
     let elem = this; // called from event handler context with this as elem
     let parent; // off-canvas elem if called from inner scrollbox
-    let up = event.pageY < elem.lastY;
+    let up = event.touches[0].pageY > elem.lastY;
     let down = !up;
-    elem.lastY = event.pageY;
```
Comment (DanielRuf) [inline js/foundation.offcanvas.js]: `event.pageY` is a valid property of jQuery events. See https://api.jquery.com/event.pagey/

File: js/foundation.offcanvas.js [hunk 2]
```diff
@@ -351,7 +351,7 @@ class OffCanvas extends Plugin {
     }
     elem.allowUp = elem.scrollTop > 0;
     elem.allowDown = elem.scrollTop < (elem.scrollHeight - elem.clientHeight);
-    elem.lastY = event.originalEvent.pageY;
+    elem.lastY = event.touches[0].pageY;
```
Comment (DanielRuf) [inline js/foundation.offcanvas.js]: `event.originalEvent` is a valid property of the jQuery event object. See https://api.jquery.com/category/events/event-object/

**Comment by DanielRuf** [comment]
Please create a codepen which reproduces the bug in Foundation Sites 6.6.1.

**Comment by rastislav-chynoransky** [comment]
@DanielRuf Hi, Daniel! You're right about the jQuery events, the `pageY` property exists, but in touch-enabled context it returns `undefined`.

I realized there are other problems with this code. 

- the checkings for `allowUp` and `allowDown` are useless, because we always allow them by offsetting of one pixel
- slow one-pixel scrolling of parent element of a scrollbox
- `event.preventDefault()` should not be even called from passive event handler (https://developers.google.com/web/updates/2017/01/scrolling-intervention)

Here's the codepen, before:
https://codepen.io/rchyno/pen/eYNgBYd
After:
https://codepen.io/rchyno/full/vYOgyyQ

The stuck-at-the-bottom problem may or may not happen, depending on whether this condition passes:
https://github.com/foundation/foundation-sites/blob/aa82df837f324205a8233bae3fa8e1ae828eba42/js/foundation.offcanvas.js#L348

(`scrollTop` in some cases may be `> elem.scrollHeight - elem.clientHeight`)

File: js/foundation.offcanvas.js [hunk 3]
```diff
@@ -336,22 +336,7 @@ class OffCanvas extends Plugin {
...
-      }
-      // If we're at the bottom, scroll up one pixel to allow scrolling down
-      if (elem.scrollTop === elem.scrollHeight - elem.clientHeight) {
-        elem.scrollTop = elem.scrollHeight - elem.clientHeight - 1;
-      }
-    }
-    elem.allowUp = elem.scrollTop > 0;
-    elem.allowDown = elem.scrollTop < (elem.scrollHeight - elem.clientHeight);
-    elem.lastY = event.originalEvent.pageY;
+    this.lastY = event.touches[0].pageY;
```
Comment (DanielRuf) [inline js/foundation.offcanvas.js]: Does that even work in non-touch context?

Comment (rastislav-chynoransky) [reply js/foundation.offcanvas.js]: The handlers are called only in touch context
https://github.com/foundation/foundation-sites/blob/aa82df837f324205a8233bae3fa8e1ae828eba42/js/foundation.offcanvas.js#L430-L433

**Comment by DanielRuf** [comment]
Would be good to have a test which fails without the changes and succeeds with the changes. Not sure if this is possible.

**Comment by rastislav-chynoransky** [comment]
@DanielRuf I'm not sure if I'm able to write those. I may help though.

**Comment by rastislav-chynoransky** [comment]
@DanielRuf I've tried to improve readability by splitting the `_stopScrollPropagation` into two -- each handling different target.

**Review by joeworkman — APPROVED** [review]

**Pull request closed without merging**